src/marten/conf/global_settings/database.cr

```diff
@@ -92,6 +92,35 @@ module Marten
         def initialize(@id : String)
         end
 
+        def from_url(url : String)
+          # URI.parse cant parse 'sqlite://:memory:'
+          if url.starts_with? "sqlite://:memory:"
+            self.backend = "sqlite"
+            self.name = ":memory:"
+            return
+          end
+
+          uri = URI.parse url
+
+          self.backend = uri.scheme
+          self.host = uri.host
+          self.port = uri.port
+          self.user = uri.user
+          self.password = uri.password
+          self.name = uri.scheme == "sqlite" ? uri.host : uri.path[1..]?
+
+          params_map = uri.query_params.to_h
+          
+          self.checkout_timeout = params_map.delete("checkout_timeout").try &.to_f64 || @checkout_timeout
```

Comment:
It would probably be safer to assign these values only when they are present in the query params:

```suggestion
          if params_map.has_key?("checkout_timeout")
            self.checkout_timeout = params_map.delete("checkout_timeout").try &.to_f64
          end
```

Reply:
Unless I'm missing something, the compiler does not like this assignment as it still thinks delete could return a `Nil`.

Reply:
I could also just deal with the deletion of these keys when setting `self.options` if that sounds better.

Reply:
> Unless I'm missing something, the compiler does not like this assignment as it still thinks delete could return a Nil.

Yes, you have to check that you are in the presence a non-nil value. For example:

```suggestion
          if !(v = params_map.delete("checkout_timeout").try &.to_f64).nil?
            self.checkout_timeout = v
          end
```